Readme.md

```diff
@@ -0,0 +1,1432 @@
+
+Constants must always be defined in objects. If a constant is only used in one class, then it should be defined in its companion object. If a constant is used in multiple classes, it should be defined in some common object.
+
+Constants must not be defined in package objects.
+
+### Getters
+
+Do not prefix getters with `get`, just name the getter after the thing it retrieves:
+
+```scala
+trait Nameable {
+  def name: String
+}
+```
+
+### Names shadowing
+
+Avoid clashes of the names with pattern matching, e.g. don't do this:
+
+```scala
+case class User(...)
+case class Organization(owner: User)
+
+val user = User(...)
+...
+organization match {
+  case Organization(user) => ...  // user clashes with the above val
+}
+```
+
+do e.g. this:
+
+```scala
+organization match {
+  case Organization(owner) => ...
+}
+```
+
+Names shadowing can be resolved by the IDEs, but nevertheless is a bad idea because it is very easy to mistake one variable for another. Use more appropriate names for each variable and pattern.
+
+## Imports
+
+### Imports order
+
+Use the IDEA imports optimizer to sort imports. **TODO: add the snippet with the imports configuration** Source code files should not contain unused imports.
+
+### Relative imports
+
+Always use absolute imports; IDEA code style configuration enforces this. Relative imports almost always decrease the readability.
+
+### Imports location
+
+Always put imports at the top of the file instead of nested scopes. In other words, imports must always be grouped together in one place.
+
+## Braces
+
+### Definitions
+
+Avoid using braces for simple expressions; write this:
+
+```scala
+def square(x: Int) = x * x
+```
+
+but not this:
+
+```scala
+def square(x: Int) = {
+  x * x
+}
+```
+
+even though it may be tempting to distinguish the method body syntactically. The first alternative has less clutter and is easier to read. Avoid syntactical ceremony unless it clarifies.
+
+### Scoping and blocks
+
+It is acceptable to use braces in order to limit the namespace pollution, but this should almost always be used only when the result of the block is assigned to another variable:
+
+```scala
+val someValue = {
+  val localVariable = ...
+  ...
+}
+```
+
+Moreover, using blocks to put complex expressions in other expressions, most notable function calls, hinders readability in a really bad way and should be avoided:
+
+```scala
+// bad
+someFunction(arg1, arg2, {
+  ...more code...
+}, arg4)
+
+// good
+val arg3 = {
+  ...more code...
+}
+someFunction(arg1, arg2, arg3, arg4)
+```
+
+## Comments
+
+Use Scaladoc to provide API documentation. Use the following style:
+
+```scala
+/**
+ * ServiceBuilder builds services 
+ * ...
+ */
+```
+
+but not the standard Scaladoc style:
+
+```scala
+/** ServiceBuilder builds services
+  * ...
+  */
```

Comment:
For me, it absolutely does not matter, which Scaladoc formatting style should be used.
I have the only suggestion: the selected formatting style should be automatically applied by the IDE code re-formatting tool.